docs/patterns/tools/serve-tools.mdx

```diff
@@ -0,0 +1,155 @@
+---
+
+<Steps>
+    <Step title="Install Required Packages">
+        First, install all necessary packages:
+
+        <CodeGroup>
+            ```bash Python Install Packages
+            pip install composio_core composio_openai
+            pip install fastapi uvicorn pydantic
+            ```
+        </CodeGroup>
+    </Step>
+
+    <Step title="Set Up Environment Variables">
+        Configure your API keys:
+
+        <CodeGroup>
+            ```bash Environment Setup
+            export COMPOSIO_API_KEY=<your-composio-api-key>
+            export OPENAI_API_KEY=<your-openai-api-key>
+            ```
+        </CodeGroup>
+
+        <Note>
+            Replace the API keys with your actual Composio and OpenAI API keys.
+        </Note>
+    </Step>
+
+    <Step title="Authenticate GitHub Account">
+        <CodeGroup>
+            ```bash Authenticate GitHub Account
+            composio add github # Launches GitHub login
+            ```
+        </CodeGroup>
+
+        <Warning>
+            If you haven't authenticated your GitHub account, complete the authentication process.
+        </Warning>
+    </Step>
+
+    <Step title="Set up FastAPI Imports">
+        Create a new file `main.py` and add the necessary imports:
+
+        <CodeGroup>
+            ```python main.py
+            from fastapi import FastAPI
+            from pydantic import BaseModel
+            from composio_openai import ComposioToolSet, App
+            from openai import OpenAI
+
+            # Initialize FastAPI app
+            app = FastAPI()
+            ```
+        </CodeGroup>
+    </Step>
+
+    <Step title="Create Request Model">
+        Define the Pydantic model for request validation:
+
+        <CodeGroup>
+            ```python main.py
+            class TaskRequest(BaseModel):
+                task: str  # This will contain the natural language task description
+            ```
+        </CodeGroup>
+
+        <Note>
+            Pydantic ensures that incoming requests contain a valid `task` field.
+        </Note>
+    </Step>
+
+    <Step title="Initialize API Clients">
+        Set up the OpenAI and Composio clients:
+
+        <CodeGroup>
+            ```python main.py
+            # Initialize clients
+            openai_client = OpenAI()
+            composio_toolset = ComposioToolSet()
+            tools = composio_toolset.get_tools(apps=[App.GITHUB])
+            ```
+        </CodeGroup>
+
+        <Note>
+            This step prepares the tools needed for GitHub interactions.
+        </Note>
+    </Step>
+
+    <Step title="Create API Endpoint">
+        Add the endpoint that will process tasks:
+
+        <CodeGroup>
+            ```python main.py
+            @app.post("/execute_task")
+            async def execute_task(request: TaskRequest):
+                response = openai_client.chat.completions.create(
+                    model="gpt-4o-mini",
```

Comment:
The model name `gpt-4o-mini` appears to be incorrect. Based on the codebase patterns and OpenAI's available models, you should use a standard model name like `gpt-4-turbo-preview` or `gpt-4-1106-preview`. Using an invalid model name will cause API calls to fail.